include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -222,11 +231,30 @@ class format_parse : public format_base
         return (std::find_if(begin_it, end_of_options_it,
             [&] (const std::string & v)
             {
-                size_t id_size{(prepend_dash(id)).size()};
-                if (v.size() < id_size)
-                    return false; // cannot be the correct identifier
+                std::string full_id = prepend_dash(id);
 
-                return v.substr(0, id_size) == prepend_dash(id); // check if prefix of v is the same
+                if constexpr (std::same_as<id_type, char>) // short id
+                {
+                    // check if prefix of v is the same since "-oValue" "-o=Value" "-o value" are all valid
+                    return v.substr(0, full_id.size()) == full_id;
+                }
+                else
+                {
+                    // only "--opt Value" or "--opt=Value" are valid
+                    if (v.substr(0, full_id.size()) == full_id)
+                    {
+                        if (v.size() == full_id.size())
+                            return true; // value was separated by space and is next in argv
+                        else if (v[full_id.size()] == '=')
+                            return true; // value is separated by '='
+                        else
+                            return false;
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
```

Comment:
```suggestion
                    return v.substr(0, full_id.size()) == full_id && (v.size() == full_id.size() || v[full_id.size()] == '=');
```
?

Reply:
Nice thanks.

Reply:
Although the line is too long so I will store a variable and return that.